kernel/tests/golden_tables.rs

```diff
@@ -0,0 +1,343 @@
+//! Golden table tests adapted from delta-io/delta tests.
+//!
+//! Data (golden tables) are stored in tests/golden_data/<table_name>.tar.zst
+//! Each table directory has a table/ and expected/ subdirectory with the input/output respectively
```

Comment:
how much work would it be to change this to be a `delta/` and an `expected/` subdir? That would match how we do it in DAT is all. Not a huge deal if recreating all the zst's is a lot of work.

Reply:
i can write a script to do this :)

Reply:
done